server/src/main/java/com/cloud/api/ApiResponseHelper.java

```diff
@@ -2213,13 +2214,75 @@ public NetworkResponse createNetworkResponse(ResponseView view, Network network)
+                            capabilityResponse.setCanChoose(true);
+                        } else {
+                            capabilityResponse.setCanChoose(false);
+                        }
                         capabilityResponse.setObjectName("capability");
                         capabilityResponses.add(capabilityResponse);
                     }
-                    serviceResponse.setCapabilities(capabilityResponses);
                 }
 
+                if (Service.SourceNat == service) {
+                    // overwrite
+                    capabilityResponses = new ArrayList<CapabilityResponse>();
+                    CapabilityResponse sharedSourceNat = new CapabilityResponse();
+                    sharedSourceNat.setName(Capability.SupportedSourceNatTypes.getName());
+                    sharedSourceNat.setValue(networkOffering.isSharedSourceNat() ? "perzone" : "peraccount");
+                    sharedSourceNat.setCanChoose(true);
+                    capabilityResponses.add(sharedSourceNat);
+
+                    CapabilityResponse redundantRouter = new CapabilityResponse();
+                    redundantRouter.setName(Capability.RedundantRouter.getName());
+                    redundantRouter.setValue(networkOffering.isRedundantRouter() ? "true" : "false");
+                    redundantRouter.setCanChoose(true);
+                    capabilityResponses.add(redundantRouter);
+
+                } else if (service == Service.StaticNat) {
+                    CapabilityResponse eIp = new CapabilityResponse();
+                    eIp.setName(Capability.ElasticIp.getName());
+                    eIp.setValue(networkOffering.isElasticIp() ? "true" : "false");
+                    eIp.setCanChoose(false);
+                    capabilityResponses.add(eIp);
+
+                    CapabilityResponse associatePublicIp = new CapabilityResponse();
+                    associatePublicIp.setName(Capability.AssociatePublicIP.getName());
+                    associatePublicIp.setValue(networkOffering.isAssociatePublicIP() ? "true" : "false");
+                    associatePublicIp.setCanChoose(false);
+                    capabilityResponses.add(associatePublicIp);
+
+                } else if (Service.Lb == service) {
+                    CapabilityResponse eLb = new CapabilityResponse();
+                    eLb.setName(Capability.ElasticLb.getName());
+                    eLb.setValue(networkOffering.isElasticLb() ? "true" : "false");
+                    eLb.setCanChoose(false);
+                    capabilityResponses.add(eLb);
+
+                    CapabilityResponse inline = new CapabilityResponse();
+                    inline.setName(Capability.InlineMode.getName());
+                    inline.setValue(networkOffering.isInline() ? "true" : "false");
+                    inline.setCanChoose(false);
+                    capabilityResponses.add(inline);
```

Comment:
I think that it is possible to extract most of the "duplicated" lines by creating a method that returns a CapabilityResponse based on `name`, `value`, and `canChoose`.

Reply:
@GabrielBrascher  please review it again. made some changes to the code